src/thorin/pass/rw/lower_for.cpp

```diff
@@ -0,0 +1,54 @@
+#include "thorin/pass/rw/lower_for.h"
+
+#include "thorin/lam.h"
+
+namespace thorin {
+
+const Def* LowerFor::rewrite(const Def* def) {
+    if (auto app = def->isa<App>())
+        if (auto for_ax = app->axiom(); for_ax && for_ax->tag() == Tag::For) {
```

Comment:
You can match that a bit simpler. Sth like this:
```cpp
if (auto for_ax = isa<Tag::For>(def)) /*...*/
```
This will give you a `Query` that lets you access the axiom via `Query::axiom` and otherwise behaves like the matched `def`.

Reply:
neat